### ISSUE TITLE
Restore "rule.id" as a clickable action in visualizations and Discover

### ISSUE THREAD

**Issue by jesusgn90** [wazuh/wazuh-kibana-app#1529]
| Wazuh | Elastic | Rev |
| ----- | ------- | --- |
| 3.9   | 7.x     | n/a |

**Description**
Before the Elastic 7 migration, the app could show to the user the field _rule.id_ as a clickable action in visualizations and in the Discover documents, once the user clicks on the value, then the _Management > Ruleset > :rule.id_ was opened.

Right now, it's missing because we disabled it until we found a solution to restore it.

**Proposal**

The field actions are defined at index pattern level, here we can see how it looks for 6.x: https://github.com/wazuh/wazuh-kibana-app/blob/34fe58c667f2f5004d22702a410cb823bd2f6a72/server/lib/elastic-wrapper.js#L312-L317.

**Comment by jesusgn90** [comment]
**Update**

First step was just adding the missing line in our index pattern:

```json
 "rule.id":{"id":"url","params":{"urlTemplate":"wazuh#/manager/?tab=ruleset&ruleid={{value}}","labelTemplate":"{{value}}","openLinkInCurrentTab":true}}, 
```

After some researching I've concluded the next things:

- Both the Kibana Discover and the Wazuh Discover are working as expected for URL fields.

![image](https://user-images.githubusercontent.com/8860227/59356719-bd437100-8d29-11e9-9d1f-2f4c55384be1.png)

- Wazuh visualizations work but are not giving the field the right format.

![image](https://user-images.githubusercontent.com/8860227/59356770-d2b89b00-8d29-11e9-93ea-d6b94af90016.png)

Then I've tried to override the specific module of Kibana for formatting fields, this would be a solution because it gives us the right format:

![image](https://user-images.githubusercontent.com/8860227/59356846-fed41c00-8d29-11e9-8af6-53c51133e022.png)

However, I think there are different solutions for this without overriding their modules, just a little thing changed for Elastic 7 and we are still looking for it. Also, I've written to the Kibana team, I hope they give us some light on this issue.

Related Kibana modules:

- The converter https://github.com/elastic/kibana/blob/v7.1.1/src/legacy/core_plugins/kibana/common/field_formats/types/url.js
- The register for the converter https://github.com/elastic/kibana/blob/v7.1.1/src/legacy/core_plugins/kibana/public/field_formats/register.js
- Tool for highlight the field https://github.com/elastic/kibana/blob/v7.1.1/src/legacy/core_plugins/kibana/common/highlight/highlight_html.js

**Comment by jesusgn90** [comment]
**Update**

After more researching, I've concluded Kibana 7.x has a bug related to relative path URLs and using it in visualizations. Since it's working for the Discover, it's not working for visualizations nor for the Wazuh app either the Kibana > Visualize itself.

I've decided to override a class property named `FieldFormat` from Kibana in the Wazuh app. 

The change was made here: https://github.com/wazuh/wazuh-kibana-app/commit/bb11d82410c8921f29bcd3b1654a986261740378

Now it detects if it's a Wazuh field expected to be a relative URL, then it creates the right URL with the proper format even if it's for a visualization.

We need to test it of course, but this change seems to be the solution so far.

![image](https://user-images.githubusercontent.com/8860227/59921351-b43f5780-942d-11e9-8704-b858ae5ec1c6.png)

The change is expected to be working for the existing URLs such as the Vulnerability absolute URLs (CVEs) and any other new relative or absolute URLs we may add in further versions.
